### DISCUSSION TITLE
Is there a way to add `updated_at` column in piccolo ?

### DISCUSSION THREAD

**Discussion by sumitsharansatsangi** [piccolo-orm/piccolo_admin#277] [Q&A] [answered]
How can I add a table as follows:
```python
class Pet(Table):
    id=UUID(primary_key=True)
    name=Varchar()
    created_at =Timestamp(default=datetime.now())
    updated_at =Timestamp(default=datetime.now())  # Not a right way 
```   

What is happening here is that `updated_at` is not updating while we update the row using piccolo_admin, the behavior is understood.

One way to do this is using [Hooks ](https://piccolo-admin.readthedocs.io/en/latest/table_config/index.html#hooks) , but if it has to be done in lots of table, Then it can be quite tedious work. 
If `piccolo` can have something like follows as in `django`, it will be helpful
```python
class Pet(Table):
    id=UUID(primary_key=True)
    name=Varchar()
    created_at =Timestamp(default=datetime.now())
    updated_at =Timestamp(default_updated=datetime.now()) 
```
Other way it can be done is using [custom datatype](https://piccolo-orm.readthedocs.io/en/latest/piccolo/schema/advanced.html#how-to-create-custom-column-types), but I don't feel comfortable  with it because I have not understood it properly.

**Comment by dantownsend** [comment, marked as answer]
Give `auto_update` a try - it should do what you need:

```python
import datetime

class MyTable(Table):
    updated_date = Timestamp(auto_update=datetime.datetime.now)
```

**Reply by sumitsharansatsangi** [reply]
Let me try, please.

**Reply by sumitsharansatsangi** [reply]
Thanks, it works as expected.
